### DISCUSSION TITLE
Multipath after already deployed some vm's

### DISCUSSION THREAD

**Discussion by gionag** [TheGrandWazoo/freenas-proxmox#161] [Q&A] [answered]
Hello !
as per title, what about enabling multipath AFTER i have already added a bunch of vms ?
it will figure it out on its own or i need to do something manual after installing/configuring multipath system-wise ?
What i have to expect ?

thanks

**Comment by TheGrandWazoo** [comment, marked as answer]
Sorry this went unanswered for so long. At the time this was asked, multipath was not supported at all. Since then, multipath shipped in **v3.2.0** as a separate, independently installable plugin (`truenas-proxmox-multipath`), distributed via its own apt component:

```
deb [signed-by=/etc/apt/keyrings/truenas-proxmox.gpg] https://thegrandwazoo.github.io/freenas-proxmox v3 main multipath
apt install truenas-proxmox-multipath
```

To answer the original question directly: yes, you can enable it after VMs are already deployed. It is a separate storage type (`truenas-multipath`) that you add alongside your existing storage — it does not retroactively convert existing single-path zvols. Existing VM disks stay right where they are; new disks (or disks you move over) go on the multipath storage. Active-active multipath with path failover has been tested and confirmed working on both TrueNAS CORE 13.x (FreeBSD/ALUA) and SCALE 24.10/25.04 — see #256 for the full test log.
